index.js

```diff
@@ -92,21 +92,33 @@ async function onTrack(event, settings) {
  * @return any
  */
 async function onIdentify(event, settings) {
-    if (!event.userId) {
-        throw new InvalidEventPayload('userId is required when identifying')
-    }
     let { properties } = event
-    properties = properties || {}
-    properties['$anon_distinct_id'] = event.anonymousId
-    return await onTrack(
-        {
-            ...event,
-            event: '$identify',
-            properties,
-            $set: event.traits,
-        },
-        settings
-    )
+    properties = {
+        ...(properties || {}),
+        $set: event.traits,
+    }
+
+    if (event.userId) {
+        properties['$anon_distinct_id'] = event.anonymousId
```

Comment:
if `event.anonymousId` is empty this will be undefined, should it be an if instead?

Reply:
I got the impression that both will be set but thats a good point. I'll change it